processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/RowBasedGrouperHelper.java

```diff
@@ -398,8 +399,10 @@ public static ColumnSelectorFactory createResultRowBasedColumnSelectorFactory(
     );
 
     final SettableSupplier<ResultRow> rowSupplier = new SettableSupplier<>();
-    final ColumnSelectorFactory columnSelectorFactory =
-        RowBasedGrouperHelper.createResultRowBasedColumnSelectorFactory(subquery, rowSupplier);
+    final ColumnSelectorFactory columnSelectorFactory = new VirtualizedColumnSelectorFactory(
+        RowBasedGrouperHelper.createResultRowBasedColumnSelectorFactory(subquery, rowSupplier),
+        query.getVirtualColumns()
+    );
```

Comment:
```suggestion
    final ColumnSelectorFactory columnSelectorFactory = query.getVirtualColumns().wrap(
        RowBasedGrouperHelper.createResultRowBasedColumnSelectorFactory(subquery, rowSupplier)
    );
```
would be more consistent with how other methods such as `createGrouperAccumulatorPair` are doing this 

Since `VirtualColumns.wrap` is currently the only 'production' method that is calling `VirtualizedColumnSelectorFactory` constructor right now, so I think we should probably change this to use `wrap`

Reply:
Done